### DISCUSSION TITLE
webhook integration

### DISCUSSION THREAD

**Discussion by gerhardpeter** [technologiestiftung/stadtpuls#4] [Q&A] [answered]
"body": "{\"statusCode\":400,\"error\":\"Bad Request\",\"message\":\"body.uplink_message.decoded_payload should have required property 'measurements'\"}",
obviously the webhook integration description is misleading.

should be corrected. Esp. the payload content and description

**Comment by ff6347** [comment]
I took a look into the docs. Sorry for the incomplete docs. To get the data into the right shape we rely on payload formatters. I didn't see it any notion of that @julizet

**Reply by ff6347** [reply]
This is an example formatter we use:

```js
function decodeUplink(input) {
  var temp = input.bytes[0] << 8 | input.bytes[1];
  var dezi = input.bytes[2];
  
  return {
    data: {
      bytes: input.bytes,
      measurements: [temp, dezi]
    },
    warnings: [],
    errors: []
  };
}
```

**Reply by gerhardpeter** [reply]
what is correct: Authorization or authorization - as shwon by TTS CE ?

**Reply by gerhardpeter** [reply]
it would be better to use MQTT

**Reply by gerhardpeter** [reply]
cannot get contact to webhook

**Reply by ff6347** [reply]
> what is correct: Authorization or authorization - as shwon by TTS CE ?
> 
> 

Both are fine I think. (Don't have a test case for that)

**Reply by ff6347** [reply]
> it would be better to use MQTT

Yes totally. We are still in beta and figuring things out ourselves. MQTT is on our "things to do" list 🙃

**Reply by ff6347** [reply]
> cannot get contact to webhook
> 

Debugging webhooks via the console is hard. Maybe we can schedule a call in the next week and you tell us a little about the experience and where everything went south?
Use my email to used before. Tuesday or Wednesday would be good for me

**Comment by gerhardpeter** [comment]
thats not really good because i have to change the original code.
it would be better to select the right label in stadtpuls.com

thx

g

Am Sa., 20. Nov. 2021 um 11:45 Uhr schrieb Fabian Morón Zirfas <
***@***.***>:

> This is an example formatter we use:
>
> function decodeUplink(input) {
>   var temp = input.bytes[0] << 8 | input.bytes[1];
>   var dezi = input.bytes[2];
>
>   return {
>     data: {
>       bytes: input.bytes,
>       measurements: [temp, dezi]
>     },
>     warnings: [],
>     errors: []
>   };}
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/technologiestiftung/stadtpuls-community/discussions/4#discussioncomment-1674299>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARYT6MHIH6AL26Z4YJIWQPTUM5325ANCNFSM5IKTPPJQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Beste Grüße,
gerhard

http://thethingsnetwork.org/c/berlin

**Comment by gerhardpeter** [comment]
i tried and got this

[image: image.png]

Am Sa., 20. Nov. 2021 um 11:45 Uhr schrieb Fabian Morón Zirfas <
***@***.***>:

> This is an example formatter we use:
>
> function decodeUplink(input) {
>   var temp = input.bytes[0] << 8 | input.bytes[1];
>   var dezi = input.bytes[2];
>
>   return {
>     data: {
>       bytes: input.bytes,
>       measurements: [temp, dezi]
>     },
>     warnings: [],
>     errors: []
>   };}
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/technologiestiftung/stadtpuls-community/discussions/4#discussioncomment-1674299>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARYT6MHIH6AL26Z4YJIWQPTUM5325ANCNFSM5IKTPPJQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Beste Grüße,
gerhard

http://thethingsnetwork.org/c/berlin

**Reply by ff6347** [reply]
I can't see the image. I guess reply's per mail don't allow attachments

**Comment by gerhardpeter** [comment]
if you have time you will reach me today
0151.xxxxxxxxxxxx

danke
gerhard

EDIT: Redacted your phone number

**Comment by ff6347** [comment, marked as answer]
The error was some whitespace in the device id
